alacritty/src/config/window.rs

```diff
@@ -74,10 +74,18 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        let is_columns_set = self.dimensions.columns.0 != 0;
+        let is_lines_set = self.dimensions.lines != 0;
+
+        // Throw warning if only one of columns, lines is set
+        if is_lines_set ^ is_columns_set {
```

Comment:
Inlining this will make things significantly simpler. The variable names are also incorrect, since `0` can be explicitly set too.